### DISCUSSION TITLE
API to download artifact

### DISCUSSION THREAD

**Discussion by tuler** [community/community#25345] [Actions]
<p>Is there an API for Github Actions?</p>
<p>I need to download an artifact produced by actions/upload-artifact.</p>
<p>I don’t mean to download the artifact for a subsequent job, like I can do with actions/download-artifact.</p>
<p>Thanks,</p>
<p>Danilo</p>

**Comment by TheBlackPlague** [comment]
<p>Bump as I’m looking for this as well.</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/tuler">@tuler</a>,</p>
<p>Thank you for being here and our apologies for the delay in response. At this time there is not, however this is very much on the roadmap and being worked on. </p>

**Comment by ilyazub** [comment]
<p>Hi <a class="mention" href="/andreagriffiths11">@andreagriffiths11</a>,</p>
<p>Thanks for your response. Can you provide ETA for this API?</p>

**Comment by vitalykarasik** [comment]
<p>+1</p>
<p>very upsetting that so cool feature released without API…</p>

**Comment by mscoutermarsh** [comment]
<p>We’re working on this now. Rough ETA is 4-6 weeks.</p>

**Comment by shridharkalagi** [comment]
<p>Is the API available now?</p>

**Comment by mscoutermarsh** [comment, marked as answer]
<p>Yes, this is now available.</p>
<p><a href="https://developer.github.com/v3/actions/artifacts/" rel="nofollow noopener">https://developer.github.com/v3/actions/artifacts/</a></p>

**Comment by shridharkalagi** [comment]
<p>Is there a way to get the latest Run id? </p>

**Comment by lelegard** [comment]
<pre><code>curl -sL https://api.github.com/repos/USER/REPO/actions/workflows/WORKFLOW.yml/runs | jq -r '.workflow_runs[0].id?'
</code></pre>
<p>With, of course, USER, REPO and WORKFLOW adjusted</p>

**Comment by jpfeuffer** [comment]
<p>The documentation is wrong. It says everyone with read access can download artifacts, but when I try the examples, it wants admin rights.</p>
<p><a href="https://developer.github.com/v3/actions/artifacts/#download-an-artifact" rel="nofollow noopener">https://developer.github.com/v3/actions/artifacts/#download-an-artifact</a></p>
<p>EDIT: Also, when I give curl an admin user, it downloads the file right away and does not respond with the “Location:” as described.</p>
<p>EDIT2: Ah ok, it is intended to download the file right away but gives the Location for potential subsequent accesses. This was not immediately clear. Still, the rights issue still persists and it would be a bit of hassle to always have an admin user (token) handy in all our CI workflows that want to use this API.</p>

**Comment by larryclaman** [comment]
<p>This is great, but no one has updated the main github documentation.  On the page <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/persisting-workflow-data-using-artifacts#sharing-data-between-workflow-runs" rel="nofollow noopener">persisting-workflow-data-using-artifacts</a> it says:<br>
<em>GitHub does not currently offer a REST API to retrieve uploaded artifacts.</em><br>
<em>If you need to access artifacts from a previous workflow run, you’ll need to store the artifacts somewhere. For example, you could run a script at the end of your workflow to store build artifacts on Amazon S3 or Artifactory, and then use the storage service’s API to retrieve those artifacts in a future workflow.</em></p>
<p>It seems like this has been superceeded; can someone from GitHub please update the documentation?</p>

**Comment by benstear** [comment]
<p><a class="mention" href="/jpfeuffer">@jpfeuffer</a> I have the same question, documentation says only read access is required but when I run the command, it says “Must have admin rights to Repository.”</p>
<p>Do you think it could be because I’m working on a forked repository and not the original one?</p>

**Comment by fabiosangregorio** [comment]
<p>Bump as I really need this too.</p>
